gallery_dl/extractor/subscribestar.py

```diff
@@ -84,44 +86,98 @@ def _login_impl(self, username, password):
             if cookie.name.startswith("auth")
         }
 
-    def _media_from_post(self, html):
+    def _extract_media(self, html, media_types):
         media = []
+        media_config = {
+            "gallery": ('data-gallery="', '"', self._process_gallery_item),
+            "attachments": (
+                'class="uploads-docs"',
+                'data-role="post-edit_form"',
+                self._process_attachment_item,
+            ),
+            "link": ('data-href="', '"', self._process_media_item),
+            "audio": ('<source src="', '" type="audio/',
+                      self._process_media_item),
+        }
```

Comment:
Here we define 4 type of content types:
 - gallery : already handled before
 - attachments : already handled before
 - link: a new type to extract links from posts bodies
 - audio: if we a post has embedded audio.

For each type, we have:
 - the detection rules begin
 - the dtection rules end
 - the function that will process the content returned